MPFramework/MPFProcess.py

```diff
@@ -70,6 +71,8 @@ def run(self):
                     #Otherwise, update with the latest main process message.
                     self._MPFLog.debug("MPFProcess {} sending update to subclass".format(self.name))
                     self.update(self.task_checker.header, self.task_checker.latest_data)
+                    if not self._process_all_updates:
+                        break
```

Comment:
This is kinda ugly, but it felt like the most idiomatic way to handle this, as python doesn't have a do/while loop (can't believe I had a valid case for one of those)